src/main/java/core/basesyntax/operation/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.PurchaseException;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    private static final int MIN_QUANTITY_LEVEL = 0;
+
+    @Override
+    public void changeFruitQuantity(String fruit, int quantity) {
+        if (Storage.reportData.get(fruit) == null) {
+            throw new PurchaseException("Can't sale \"" + fruit
+                    + "\" because there are no such fruit in storage");
+        }
+        int newQuantity = Storage.reportData.get(fruit) - quantity;
+        if (newQuantity < MIN_QUANTITY_LEVEL) {
+            throw new PurchaseException("Can't sale " + quantity + " of "
```

Comment:
Use `String.format()` for such long concatenations, it would be easier to read and edit the message.